### DISCUSSION TITLE
[Resolved] Is anyone working on a SQLTools driver for DuckDB?

### DISCUSSION THREAD

**Discussion by aaronsteers** [duckdb/duckdb#5067] [General]
UPDATE:

It's alive!! 

From @RandomFractals 🙏 

https://marketplace.visualstudio.com/items?itemName=RandomFractalsInc.duckdb-sql-tools

----

DuckDB would be an amazing addition to supported DBs for the very handy VS Code [SQLTools](https://marketplace.visualstudio.com/items?itemName=mtxr.sqltools) extension. Is anyone working on this as of yet? Anyone interested?

Mirror posted there also:

- https://github.com/mtxr/vscode-sqltools/issues/1018

How-to docs:

- https://vscode-sqltools.mteixeira.dev/en/contributing/support-new-drivers/

**Comment by hannes** [comment]
Not aware of anything but who knows what's out there

**Comment by RandomFractals** [comment]
someone started working on it :)

![duckdb-sql-tools](https://user-images.githubusercontent.com/656833/212547288-ebee7769-f2aa-47a7-915d-42d03d363982.gif)

... https://twitter.com/TarasNovak/status/1614629029642133504

**Comment by RandomFractals** [comment]
Preview version of DuckDB Sql Tools VSCode extension is now available in marketplace:

https://marketplace.visualstudio.com/items?itemName=RandomFractalsInc.duckdb-sql-tools

**Reply by aaronsteers** [reply]
Stoked to see this, @RandomFractals! 🚀
I'm promoting in our Meltano Slack (https://meltano.com/slack) here: https://meltano.slack.com/archives/CFG3C3D1Q/p1674880828353659?thread_ts=1674880828.353659&cid=CFG3C3D1Q

Hopefully will get you some user feedback soon. :)

**Comment by RandomFractals** [comment]
There is also a [#DuckDBProTools](https://twitter.com/hashtag/DuckDBProTools?src=hashtag_click) version now available to our Pro sponsors on github with advanced DuckDB Objects Tree View, DuckDB v0.7.1 storage format support, and alpha version of write access mode. See our DuckDB Pro Tools v1.1.0 [announcement](https://twitter.com/TarasNovak/status/1653144891197718533) on twitter.

![duckdb-pro-tools-v1 1 0-release](https://user-images.githubusercontent.com/656833/236201303-4240dbd5-637b-4b50-8a8c-085c92f4c763.png)

**Comment by fordmustang5l** [comment]
I have two issues now.

1) random-fractal's pro tools are no longer found in github :( ?
2)  SQLite3 Editor  https://marketplace.visualstudio.com/items?itemName=yy0931.vscode-sqlite3-editor supports duckdb, but there is an issue when connecting to a database in which schemas are created (e.g. "create schema blah"

Anyone have a current solution for the VS Code plugins?

**Comment by RandomFractals** [comment]
@fordmustang5l we made the free DuckDB Sql Tools VS Code extension github repository public for now:

https://github.com/RandomFractals/duckdb-sql-tools

The other pro tools like DuckDB Pro Tools, PRQL Pro Tools, and new Markdown SQL Pro Tools we published this year require Pro sponsorship on github and for private beta collaborators only. They were never public, and will remain private for our Pro sponsors on github for now. You can sign up below using Pro or Markdown Sql Pro Tools tiers to get a collaborator invite and access them:

https://github.com/sponsors/RandomFractals

**Comment by RandomFractals** [comment]
@fordmustang5l We published updated DuckDB SQL Tools *Preview* extension version with DuckDB v0.7.1 support to VS Code Marketplace:

https://marketplace.visualstudio.com/items?itemName=RandomFractalsInc.duckdb-sql-tools

Also, see new Pro Data Tools section in docs for the latest DuckDB v0.8 support info and other custom PRQL/SQL Pro Data Tools for VS Code IDE we've been working on lately:  

https://github.com/RandomFractals/duckdb-sql-tools#pro-data-tools
